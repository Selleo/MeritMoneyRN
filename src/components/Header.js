import React, { Component } from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { white, secondaryLight, primary } from 'src/styles/colors'
import Icon from 'src/components/Icon'
import LinearGradient from 'react-native-linear-gradient'

export default class Header extends Component {
  render() {
    return (
      <LinearGradient
        colors={[secondaryLight, primary]}
        end={{ x: 1, y: 1 }}
        locations={[0, 1]}
        start={{ x: 0, y: 1 }}
        style={styles.container}
      >
        <View style={styles.headerContainer}>
          <View style={styles.textContainer}>
            <Icon name="star-empty" size={18} style={styles.star} />
            <Text style={styles.text}>15 left</Text>
          </View>
          <View style={styles.textContainer}>
            <Text style={styles.text}>24(54)</Text>
            <Text>^</Text>
          </View>
        </View>
      </LinearGradient>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    height: 70,
  },
  headerContainer: {
    paddingVertical: 25,
    paddingHorizontal: 20,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  text: {
    fontSize: 16,
    color: white,
  },
  textContainer: {
    flexDirection: 'row',
  },
  star: {
    color: white,
    paddingHorizontal: 5,
  },
})
