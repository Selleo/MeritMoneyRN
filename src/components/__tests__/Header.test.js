import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Header from '../Header'

describe('<Header />', () => {
  const defaultProps = {}

  const wrapper = renderer.create(<Header {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
