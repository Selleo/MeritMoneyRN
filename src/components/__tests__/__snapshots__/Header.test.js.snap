// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282147443,
      4281317757,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  style={
    Object {
      "height": 70,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Object {
              "color": "#ffffff",
              "paddingHorizontal": 5,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
          }
        }
      >
        15 left
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
          }
        }
      >
        24(54)
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Object {
              "color": "#ffffff",
              "paddingHorizontal": 5,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</BVLinearGradient>
`;
